### ISSUE TITLE
can't access emitted filename if that file was emitted in a file that only exists in manualChunk

### ISSUE THREAD

**Issue by sapphi-red** [rollup/rollup#4746]
### Rollup Version

3.7.2

### Operating System (or Browser)

stackblitz

### Link To Reproduction

https://stackblitz.com/edit/rollup-template-9nc4ay?file=rollup.config.js

### Expected Behaviour

I expect the filename can be obtained without an error happening.

This config is a weird one. `src/foo.js` is specified in `manualChunks` but cannot be referenced from the entrypoint (`src/main.js`).
I guess this is not intended to be possible. So alternatively, I expect the build fails with an error that tells `manualChunks` is incorrectly configured.

(I couldn't give a well summarized title)

### Actual Behaviour

The following error happens.
```
[!] (plugin debug) RollupError: Plugin error - Unable to get file name for unknown file "8d016e3f".
RollupError: Plugin error - Unable to get file name for unknown file "8d016e3f".
    at error (file:///home/projects/rollup-template-9nc4ay/node_modules/rollup/dist/shared/rollup.js:210:30)
    at FileEmitter.getFileName (file:///home/projects/rollup-template-9nc4ay/node_modules/rollup/dist/shared/rollup.js:23559:24)
    at Object.renderChunk (file://file:///home/projects/rollup-template-9nc4ay/rollup.config.js:39:39)
    at eval (file:///home/projects/rollup-template-9nc4ay/node_modules/rollup/dist/shared/rollup.js:23915:40)
```

Could be related to #4744?

Original issue: https://github.com/vitejs/vite/issues/11293

**Comment by lukastaegert** [comment]
Turns out the issue here is that we have separate handlers for emitted files between build and generate phase. The reason is that emitted files in one output should not be emitted in another output. So we have one central file-emission-handler during build, and when outputs are generated, all emitted files are copied over from the central handler to the output specific handler.
How does this relate to this issue? In your example, `manualChunks` triggers the `transform` hook of a module for the first time while an output is generated. The `transform` hook, however, relates to the module graph, which is a shared resource between outputs, and therefore uses the shared handler. However, new emissions are currently not copied over again to the outputs.

Now the question is, what should the solution be? Unfortunately, Rollup does not know why a file was emitted. Maybe the file really is needed, who knows? My current ideas for a solution go into the direction of allowing to keep emitting files via `transform` hooks from the shared handler and then directly copying the file to all outputs. This may not really be "correct" in this case (well, in this case you would not want the file at all), but one idea for a long term solution would be to add an additional flag when emitting assets like "removeIfUnused" that only keeps an emitted file in the output if it is referenced from any chunk. Then in our example, the file would still not be emitted, but we could in general avoid more unnecessary emitted files (think an SVG import that is not actually used etc.)

Will follow up on this one.

**Comment by lukastaegert** [comment]
Fix at #4761

**Comment by sapphi-red** [comment]
Thanks! I tested out that PR and worked as I expected.
